lib/cmds/update.ts

```diff
@@ -251,3 +271,62 @@ function installAppium(binary: Binary, outputDir: string): void {
       path.resolve(folder, 'package.json'), JSON.stringify({scripts: {appium: 'appium'}}));
   spawn('npm', ['install', 'appium@' + binary.version()], null, {cwd: folder});
 }
+
+interface BinaryPath {
+  last?: string, all?: string[]
+}
+
+interface BrowserFile {
+  chrome?: BinaryPath, standalone?: BinaryPath, gecko?: BinaryPath, iedriver?: BinaryPath
+}
+
+function updateBrowserFile<T extends Binary>(binary: T, outputDir: string) {
+  let currentDownload = path.resolve(outputDir, binary.executableFilename(Config.osType()));
+
+  // if browserFile[id] exists, we should update it
+  if ((browserFile as any)[binary.id()]) {
+    let binaryPath: BinaryPath = (browserFile as any)[binary.id()];
+    if (binaryPath.last === currentDownload) {
+      return;
+    } else {
+      binaryPath.last = currentDownload;
+      for (let bin of binaryPath.all) {
+        if (bin === currentDownload) {
+          return;
+        }
+      }
+      binaryPath.all.push(currentDownload);
+    }
+  } else {
+    // the browserFile[id] does not exist and we should create the entry.
+    let versions: string[] = [];
+    let downloaded = FileManager.downloadedBinaries(outputDir);
+    if (!downloaded[binary.id()] || !downloaded[binary.id()].versions) {
```

Comment:
Won't this always be the case?  If `(browserFile as any)[binary.id()]` was undefined, that means `update-config.json` didn't have the browser, which means it hasn't been installed yet